tests/test_s3blobstore.py

```diff
@@ -145,5 +149,22 @@ def test_get_bucket_region(self):
         self.assertEqual(handle.get_bucket_region(self.test_us_east_1_bucket), 'us-east-1')
         self.assertNotEqual(handle.get_bucket_region(self.test_non_us_east_1_bucket), 'us-east-1')
 
+    def test_raises_read_error(self):
+        """
+        Ensure that we handle botocore ReadTimeouts
```

Comment:
Also, this only tests connection failures.  What about read failures?  Ideally, we should change the host that boto connects to, and verify the connection failures that way.

Reply:
Added a proxy-server based test for read failure.
Use incorrect host for connection failure.